client/state/sites/plans/selectors.js

```diff
@@ -22,13 +23,43 @@ export function getPlansBySiteId( state, siteId ) {
 	return state.sites.plans[ siteId ] || initialSiteState;
 }
 
-export function hasDomainCredit( state, siteId ) {
+export const getCurrentPlan = ( state, siteId ) => {
 	const plans = getPlansBySiteId( state, siteId );
 	if ( plans.data ) {
 		const currentPlan = find( plans.data, 'currentPlan' );
-		return currentPlan.hasDomainCredit;
+
+		if ( currentPlan ) {
+			return currentPlan;
+		}
+
+		const site = getSite( state, siteId );
+
+		if ( ! site.plan ) {
+			return null;
+		}
+
+		// does site enterprise plan?
+		if ( site.plan.product_id === 1007 ) {
+			// lets assign business plan for old enterprise plan
+			return find( plans.data, { productSlug: 'business-bundle' } );
+		}
+
+		return site.plan;
 	}
+
 	return null;
+};
+
+export function hasDomainCredit( state, siteId ) {
+	if ( ! siteId ) {
+		return initialSiteState;
+	}
+	const currentPlan = getCurrentPlan( state, siteId );
+	if ( ! currentPlan ) {
+		return null;
+	}
+
+	return currentPlan.hasDomainCredit;
```

Comment:
These last few lines could probably be served a bit clearer with a defaulting get.

```js
const currentPlan = getCurrentPlan( state, siteId );

return get( currentPlan, 'hasDomainCredit', null );
```

Of course, if `undefined` is fine, you could leave off that defaulting `null`